src/Language/PureScript/Docs/Convert.hs

```diff
@@ -38,6 +102,95 @@ convertModule m@(P.Module _ coms moduleName  _ _) =
     >>> augmentDeclarations
     >>> map addDefaultFixity
 
+updateReExports ::
+  (Functor m, Applicative m,
+  MonadState (Map P.ModuleName (P.Module, Module)) m) =>
+  P.Env ->
+  [P.ModuleName] ->
+  m ()
+updateReExports env order = do
+  void $ traverse go order
+  where
+  go mn = do
+    (ast, mdl) <- lookup' mn
+    mdl' <- expandReExports env ast mdl
+    modify (Map.insert mn (ast, mdl'))
+
+  lookup' key = do
+    v <- gets (Map.lookup key)
+    case v of
+      Just v' ->
+        pure v'
+      Nothing ->
+        P.internalError ("Language.PureScript.Docs.Convert: missing key: " ++
+                        P.runModuleName key)
+
+-- We require that modules have already been sorted (P.sortModules) in order to
+-- ensure that by the time we convert a particular module, all its dependencies
+-- have already been converted.
+expandReExports ::
+  (Functor m, Applicative m,
+  MonadState (Map P.ModuleName (P.Module, Module)) m) =>
+  P.Env ->
+  P.Module ->
+  Module ->
+  m Module
+expandReExports env ast@(P.Module _ _ mn' _ mexports) mdl = do
+  let exports = fromMaybe (err "exports were not elaborated") mexports
+  let reExportNames = mapMaybe takeReExport exports
```

Comment:
I'm not worried about the number of headings; I arrived at this design because I think it's the best way of presenting this information.

Suppose we have a ByteString type defined in Data.ByteString.Internal, re-exported by Data.ByteString, and then re-exported again by some module A. In that case, I think we would want to write in A's docs that ByteString is re-exported from Data.ByteString. More generally, I think the documentation should present things the same way as they are in the source, because I think that's more likely to make sense to readers of the docs.

The reason for the call to sortModules in this code is to make sure dependencies have already been rendered; this way, we can include the whole declaration for re-exports, including comments, so you wouldn't have to follow the chain back to the source.